From 128c8cce3415839a3d19a5cb877fd77af2538d35 Mon Sep 17 00:00:00 2001
From: Adam Majer <amajer@suse.de>
Date: Tue, 6 Mar 2018 13:46:31 +0100
Subject: [PATCH] Add support for libical 3.x (#465)

In libical3, deprecated icaltime_from_timet is removed
Use icaltime_from_timet_with_zone instead. This function
exists in libical 2.x so no need to change requirements
---
 src/ical.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ical.cc b/src/ical.cc
index e4b48511..ef069124 100644
--- a/src/ical.cc
+++ b/src/ical.cc
@@ -49,13 +49,13 @@ struct ical_event *add_event(struct ical_event *listend, icalcomponent *new_ev)
 	icaltimetype start;
 
 	start = icalcomponent_get_dtstart(new_ev);
-	if(icaltime_compare(start, icaltime_from_timet(time(NULL), 0)) <= 0) {
+	if(icaltime_compare(start, icaltime_from_timet_with_zone(time(NULL), 0, NULL)) <= 0) {
 		icalproperty *rrule = icalcomponent_get_first_property(new_ev, ICAL_RRULE_PROPERTY);
 		if(rrule) {
 			icalrecur_iterator* ritr = icalrecur_iterator_new(icalproperty_get_rrule(rrule), start);
 			icaltimetype nexttime = icalrecur_iterator_next(ritr);
 			while (!icaltime_is_null_time(nexttime)) {
-				if(icaltime_compare(nexttime, icaltime_from_timet(time(NULL), 0)) > 0) {
+				if(icaltime_compare(nexttime, icaltime_from_timet_with_zone(time(NULL), 0, NULL)) > 0) {
 					start = nexttime;
 					break;
 				}
